functions/middleware/logger.js

```diff
@@ -0,0 +1,25 @@
+require('express-async-errors');
+const {createLogger, transports} = require('winston');
+const winston = require('winston');
+
+process.on('unhandledRejection', (err) => {
+    throw err;
+});
+
+const logger = createLogger({
+    level: 'debug',
+    transports: [
+        new transports.Console({
```

Comment:
Just a placeholder: We should check if the logs show up in functions logs in firebase console

Reply:
I checked. winston logs do not show up in functions log. They have created another winston package. But did not tried it yet. If this not necessary we can just use console.log.

Reply:
I have removed it. It now returns
{
    "message": "Requested product sdfsd is not present in Firestore."
}

without status. code